examples/bank/bank.go

```diff
@@ -93,18 +96,84 @@ func (bank *Bank) moveMoney(from, to []byte, amount int64) bool {
 			return errRead
 		}
 		// Update both accounts.
-		txn.Prepare(client.Put(proto.Key(from), []byte(fmt.Sprintf("%d", fromValue-amount))))
-		txn.Prepare(client.Put(proto.Key(to), []byte(fmt.Sprintf("%d", toValue+amount))))
+		txn.Prepare(client.Put(proto.Key(from), []byte(fmt.Sprintf("%d", fromValue-amount))),
+			client.Put(proto.Key(to), []byte(fmt.Sprintf("%d", toValue+amount))))
 		return nil
 	})
 	if err != nil {
 		log.Fatal(err)
 	}
-	atomic.AddInt32(&bank.numTransactions, 1)
+	atomic.AddInt32(&bank.numTransfers, 1)
+	return true
+}
+
+// moveMoneyWithoutTransaction() moves an amount between two accounts without the
+// use of a transaction, if the amount is available in the from account.
+// Returns true on success.
+func (bank *Bank) moveMoney(from, to []byte, amount int64) bool {
+	// Early exit when from == to
+	if bytes.Compare(from, to) == 0 {
+		return true
+	}
+	fromGet := client.Get(proto.Key(from))
+	toGet := client.Get(proto.Key(to))
+	if err := bank.kvClient.Run(fromGet, toGet); err != nil {
+		log.Fatal(err)
+		return false
+	}
+	// Read from value.
+	fromValue, err := readInt64(fromGet)
+	if err != nil {
+		log.Fatal(err)
+		return false
+	}
+	// Ensure there is enough cash.
+	if fromValue < amount {
+		return false
+	}
+	// Read to value.
+	toValue, errRead := readInt64(toGet)
+	if errRead != nil {
+		log.Fatal(errRead)
+		return false
+	}
+	// Update both accounts.
+	bank.kvClient.Run(client.Put(proto.Key(from), []byte(fmt.Sprintf("%d", fromValue-amount))),
+		client.Put(proto.Key(to), []byte(fmt.Sprintf("%d", toValue+amount))))
+	atomic.AddInt32(&bank.numTransfers, 1)
 	return true
 }
 
 // Read the balances in all the accounts and return them.
+func (bank *Bank) readAllAccountsWithScan() []int64 {
+	balances := make([]int64, bank.numAccounts)
+	txnOpts := &client.TransactionOptions{Name: "Reading all balances"}
+	err := bank.kvClient.RunTransaction(txnOpts, func(txn *client.Txn) error {
+		call := client.Scan(proto.Key(makeAccountID(0)), proto.Key(makeAccountID(bank.numAccounts)), int64(bank.numAccounts))
+		if err := txn.Run(call); err != nil {
+			log.Fatal(err)
+		}
+		// Copy responses into balances.
+		scan := call.Reply.(*proto.ScanResponse)
+		if len(scan.Rows) != bank.numAccounts {
+			log.Fatal(fmt.Errorf("Could only read %d of %d rows of the database.\n", len(scan.Rows), bank.numAccounts))
```

Comment:
It's a bit strange to generate an error (`fmt.Errorf`) only to log it. Better to just use `log.Fatalf`.